### DISCUSSION TITLE
Does power allocation rollup through connected devices?

### DISCUSSION THREAD

**Discussion by signifywinter** [netbox-community/netbox#8993] [Q&A]
Is power allocation supposed to roll-up through connected devices?  i.e.,

Panel  <- Feed <- PDU Device <- Standalone PSU Device <- Node Device (500W)

In this case I know that the node device should have an allocation of 500W.  All of the other devices that connect up the chain would then add this known allocation to their own respective allocations (so the PSU, PDU, Feed, and Panel would all have 500W + any other connections).

It would seem to me that this is how it works based on not being able to set utilization of the panel itself.  Yet the values are not flowing.  Is there a depth limitation or some kind of other parameter that needs to be set for it to roll-up correctly?

Thanks in advance!

**Comment by joaolucasmacedo** [comment]
I noticed the same thing. It would be nice if were able to trace all power allocation though PDU devices.

**Comment by signifywinter** [comment]
Additionally, is power only calculated on a device if it is connected to a feed?  I have a PDU coming off a UPS that actually connects to the wall and its power does not calculate.  Similarly, I have equipment connected to power supplies with multiple nodes attached and the PSU's power utilization does not calculate.

**Reply by crzdmniac** [reply]
I noticed the same thing and assumed I was doing something wrong. The design seems to only allow for one layer deep from the feed. I plan to just add power outlets to my UPS devices to represent the PDU, I'm not sure how else to do it; i.e. "Grp 1 Outlet 1 PDU-A 1". Someone correct me if I'm wrong here. I'm not a programmer, and I'm fairly new to Netbox.
